common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleClient.java

```diff
@@ -117,6 +118,12 @@ public void fetchBlocks(
     }
   }
 
+  @Override
+  public MetricSet shuffleMetrics() {
+    checkInit();
```

Comment:
not related to the change here -- but should we also checkInit in the `close()` function?

Reply:
Seems it should be, but looks like we never touch this issue before.